svm/src/account_loader.rs

```diff
@@ -531,11 +547,11 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
                     }
                     accumulate_and_check_loaded_account_data_size(
                         &mut accumulated_accounts_data_size,
-                        owner_account.data().len(),
+                        owner_size,
                         compute_budget_limits.loaded_accounts_bytes,
                         error_metrics,
                     )?;
-                    accounts.push((*owner_id, owner_account));
+                    validated_loaders.insert(*owner_id);
```

Comment:
github won't let me comment further down - curious if we should add a (debug) assert to make sure that `accounts` length matches the message's `account_keys.len()` - we have a few things that now rely on that assumption, which was not true in the past. Slightly concerned maybe someone changes it in future and breaks stuff and we commit some extra appended accounts.